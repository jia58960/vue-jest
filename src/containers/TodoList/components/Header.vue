<template>
  <div class="header">
    <div class="header-content">
      TODO：
      <input type="text"
      class="todo-input"
      v-model="inputVal"
      @keyup.enter="addTodoItem"
      placeholder="请输入待办事项">
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data () {
    return {
      inputVal: ''
    }
  },
  methods: {
    addTodoItem () {
      const realVal = this.inputVal.trim()
      const len = realVal.length
      if (realVal && len >= 4 && len <= 10) {
        this.$emit('add', this.inputVal)
        this.inputVal = ''
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.header
{
  line-height: 60px;
  height: 60px;
  background-color:#666;
  .header-content
  {
    width: 600px;
    margin: 0 auto;
    color: #FFF;
    .todo-input
    {
      float: right;
      margin-top:16px;
      font-size:16px;
      outline: none;
    }
  }
}
</style>
