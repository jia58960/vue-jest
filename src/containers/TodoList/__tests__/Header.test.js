import {
  shallowMount
} from '@vue/test-utils'
import Header from '../components/Header.vue'

describe('Header 组件测试', () => {
  test('保存Header组件快照', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })

  test('输入框值为空，按enter不能提交。', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('.todo-input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  test('输入空格不能提交', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('.todo-input')
    input.setValue('  ')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  test('输入的字符小于10时不派发事件', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('.todo-input')
    input.setValue('abcdefghj')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
  })

  test('输入的字符小于4时不派发事件', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('.todo-input')
    input.setValue('abc')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  test('输入满足条件的字符串，按enter键发射emit事件，且将输入框值置空。', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('.todo-input')
    input.setValue('Ethan')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.$data.inputVal).toBe('')
  })
})
