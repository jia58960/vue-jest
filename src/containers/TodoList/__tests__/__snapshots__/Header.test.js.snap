// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件测试 保存Header组件快照 1`] = `
<div class="header">
  <div class="header-content">
    TODO：
    <input type="text" placeholder="请输入待办事项" class="todo-input"></div>
</div>
`;

exports[`保存Header组件快照 1`] = `
<div class="header">
  <div class="header-content">
    TODO：
    <input type="text" placeholder="请输入待办事项" class="todo-input"></div>
</div>
`;
